src/ICS26Router.sol

```diff
@@ -123,18 +123,30 @@ contract ICS26Router is IICS26Router, IBCStore, Ownable, IICS26RouterErrors, Ree
             value: abi.encodePacked(commitmentBz)
         });
 
-        ics02Client.getClient(msg_.packet.destChannel).membership(membershipMsg);
-        if (msg_.packet.timeoutTimestamp <= block.timestamp) {
-            revert IBCInvalidTimeoutTimestamp(msg_.packet.timeoutTimestamp, block.timestamp);
+        // solhint-disable-next-line no-empty-blocks
+        try ics02Client.getClient(msg_.packet.destChannel).membership(membershipMsg) { }
+        catch (bytes memory reason) {
+            revert IBCMembershipProofVerificationFailed(msg_.packet, membershipMsg, reason);
         }
 
-        bytes memory ack =
-            app.onRecvPacket(IIBCAppCallbacks.OnRecvPacketCallback({ packet: msg_.packet, relayer: msg.sender }));
-        if (ack.length == 0) {
-            revert IBCAsyncAcknowledgementNotSupported();
+        uint64 timeoutTimestamp = msg_.packet.timeoutTimestamp;
+        if (!ICS24Host.isTimestampInSeconds(timeoutTimestamp)) {
```

Comment:
Yes, I'd rather have this removed as we want to value simplicity above everything else. Putting a restriction on the user is not problematic (since frontends should take care of it). Also, even if the user sends a bad packet, it'd just timeout.

Reply:
I've reverted the nanosecond changes, if this becomes an issue in the future, I'd consider only using nanoseconds (instead of seconds). But supporting both is too much unneeded complexity and additional security considerations.

Reply:
Alright, makes sense. After having dealt with this for a bit, I feel like the devUX is pretty clunky when there are different timestamps. Unless we can change ibc-go, I would suggest in the future that we seriously consider just using nanoseconds everywhere. But for now, this works :)